lib-api/src/main/java/org/ergoplatform/appkit/babelfee/BabelFeeOperations.java

```diff
@@ -0,0 +1,259 @@
+     *                      to spend
+     * @param tokenId       tokenId to create the babel fee box for
+     * @param pricePerToken the price per raw token value
+     * @return prepared transaction to create the new babel fee box
+     */
+    public static UnsignedTransaction createNewBabelContractTx(
+        BoxOperations boxOperations,
+        ErgoId tokenId,
+        long pricePerToken
+    ) {
+        BabelFeeBoxState babelFeeBoxState = BabelFeeBoxState.newBuilder().withBoxCreator(boxOperations.getSenders().get(0))
+            .withPricePerToken(pricePerToken)
+            .withTokenId(tokenId)
+            .withValue(boxOperations.getAmountToSpend())
+            .build();
+
+        return boxOperations.buildTxWithDefaultInputs(txB -> {
+            OutBox outBox = babelFeeBoxState.buildOutbox(txB, null);
+            txB.outputs(outBox);
+            return txB;
+        });
+    }
+
+    public static UnsignedTransaction cancelBabelFeeContract(BoxOperations boxOperations, InputBox babelBox) {
+        BabelFeeBoxState babelFeeBoxState = new BabelFeeBoxState(babelBox);
+
+        Address creatorAddress = Address.fromSigmaBoolean(babelFeeBoxState.getBoxCreator().getSigmaBoolean(),
+            boxOperations.getBlockchainContext().getNetworkType());
+
+        // if we can't pay tx fee from the babel box content, we need to load another box
+        List<InputBox> boxesToSpend;
+        if (babelFeeBoxState.getValue() < boxOperations.getFeeAmount() + Parameters.MinChangeValue)
+            boxesToSpend = boxOperations.loadTop();
+        else
+            boxesToSpend = new ArrayList<>();
+
+        boxesToSpend.add(0, babelBox);
+
+        UnsignedTransactionBuilder txB = boxOperations.getBlockchainContext().newTxBuilder();
+
+        OutBoxBuilder outBoxBuilder = txB.outBoxBuilder()
+            .value(Math.max(Parameters.MinChangeValue, babelFeeBoxState.getValue() - boxOperations.getFeeAmount()))
+            .contract(creatorAddress.toErgoContract());
+
+        if (!babelBox.getTokens().isEmpty())
+            outBoxBuilder.tokens(babelBox.getTokens().get(0));
+
+        return txB.boxesToSpend(boxesToSpend)
+            .fee(boxOperations.getFeeAmount())
+            .sendChangeTo(creatorAddress)
+            .outputs(outBoxBuilder.build())
+            .build();
+
+    }
+
+    /**
+     * Tries to fetch a babel fee box from blockchain data source with the given unspent boxes loader
+     *
+     * @param ctx       current blockchain context
+     * @param loader    loader to receive unspent boxes
+     * @param tokenId   tokenId offered to swap
+     * @param feeAmount nanoerg amount needed to swap
+     * @return babel fee box satisfying the needs, or null if none available
+     */
+    @Nullable
+    public static InputBox findBabelFeeBox(BlockchainContext ctx, BoxOperations.IUnspentBoxesLoader loader, ErgoId tokenId, long feeAmount) {
```

Comment:
this method is not used

Reply:
I can provide a mocked test but this is some kind of white box testing, that's why I skipped. Should I add one?

Reply:
yes, the library code should be covered

Reply:
done